### PR TITLE
DT-1966 Support Sub court codes

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsregisterstonomisupdate/services/CourtRegisterUpdateService.kt [hunk 1]
```diff
@@ -37,43 +37,72 @@ class CourtRegisterUpdateService(
...
-        if (currentCourtDataInPrisonSystem != null) translateToSync(currentCourtDataInPrisonSystem) else null
-      val newCourtData = mergeIds(updatedCourtDataFromRegister, currentCourtDataToCompare)
+  private fun buildCourts(courtDto: CourtDto): List<CourtDataToSync> {
 
-      val diffs = checkForDifferences(currentCourtDataToCompare, newCourtData)
-      if (!diffs.areEqual()) {
-        log.info("$courtId: APPLY CHANGES=$applyChanges - Updating Prison System with court data. Changes {}", diffs)
+    if (courtDto.buildings.size < 2) {
+      return listOf(convertToPrisonCourtData(courtDto, courtDto.buildings))
+    }
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsregisterstonomisupdate/services/CourtRegisterUpdateService.kt:59]: so I guess even if a single building incorrectly has a subcode, we will ignore and just treat it as an address. Which is good.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsregisterstonomisupdate/services/CourtRegisterUpdateService.kt [hunk 2]
```diff
@@ -275,7 +304,7 @@ class CourtRegisterUpdateService(
           postalCode = building.postcode,
           county = getRefCode("COUNTY", building.county),
           country = getRefCode("COUNTRY", building.country),
-          primary = courtDto.buildings.size == 1 || building.subCode != null,
+          primary = building == buildings[0], // first one in the list?
           noFixedAddress = false,
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsregisterstonomisupdate/services/CourtRegisterUpdateService.kt:308]: would it be first with no subcode?

**Review by andymarke — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsregisterstonomisupdate/services/CourtRegisterUpdateService.kt (hunk 2 above)
Comment (Mjwillis) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsregisterstonomisupdate/services/CourtRegisterUpdateService.kt:308]: No, ones with sub codes will be be in a different object

**Pull request merged**